TreeDB/collections/column_physical_asset.go

```
@@ -163,40 +193,75 @@ func validateTrustedFloat32ProjectionRetainedJSON(ids [][]byte, projection *trus
+				return nil
+			}); err != nil {
+				return nil, err
+			}
+			for i, col := range projection.columns {
+				if err := validateTypedRetainedColumn(trimmed, col.Path, paths[i]); err != nil {
+					return nil, err
+				}
+			}
+		} else if _, _, _, err := jsonparser.Get(trimmed, projection.column); err == nil {
 			return nil, fmt.Errorf("collections: validated float32 projection retained JSON row %d contains declared column %q", row, projection.column)
 		} else if err != jsonparser.KeyPathNotFoundError {
```

golangci-lint (2.13.2)

[error] comparing with != will fail on wrapped errors. Use errors.Is to check for a specific error

(errorlint)